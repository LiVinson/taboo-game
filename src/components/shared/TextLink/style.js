
import { Link } from "react-router-dom"
import styled from "styled-components"

export const StyledTextLink = styled(Link)`

    transition: all .2s ease-in;
  &:link,
  &:visited {
    color: currentColor;
    text-decoration: none;
    transform: translateY(0)

  }

  &:focus {
    text-shadow: 0px 0.2rem 0.3rem rgba(0, 0, 0, 0.7);
    transform: translateY(-3px);
  }

  &:hover {
    filter: brightness(1.2);
  }
`

StyledTextLink.displayName="StyledTextLink";

