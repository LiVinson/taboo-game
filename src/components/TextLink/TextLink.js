import React from "react"
import { StyledTextLink } from "./style"

const TextLink = (props) => <StyledTextLink {...props}>{props.text}</StyledTextLink>

export default TextLink
