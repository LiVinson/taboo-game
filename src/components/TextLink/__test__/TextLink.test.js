import React from "react"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import TextLink from "../index"

test("TextLink renders correctly", () => {
  const props = {
    to: "/home",
    text: "I'm a Link",
  }
  const wrapper = renderer
    .create(
      <MemoryRouter>
        <TextLink {...props} />
      </MemoryRouter>
    )
    .toJSON()

  expect(wrapper).toMatchSnapshot()
})
