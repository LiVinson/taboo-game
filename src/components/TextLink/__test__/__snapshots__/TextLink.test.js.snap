// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink renders correctly 1`] = `
.c0 {
  -webkit-transition: all .2s ease-in;
  transition: all .2s ease-in;
}

.c0:link,
.c0:visited {
  color: currentColor;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c0:focus {
  text-shadow: 0px 0.2rem 0.3rem rgba(0,0,0,0.7);
  -webkit-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
}

.c0:hover {
  -webkit-filter: brightness(1.2);
  filter: brightness(1.2);
}

<a
  className="c0"
  href="/home"
  onClick={[Function]}
  text="I'm a Link"
>
  I'm a Link
</a>
`;
